habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py

```diff
@@ -167,76 +174,69 @@ def reset(self, episode: Episode):
         self._nav_to_info = self._generate_nav_start_goal(
             episode, nav_to_pos, start_hold_obj_idx=start_hold_obj_idx
         )
-        if self._pick_init:
-            # spawn agent close to and oriented to pick object for pick testing, [wip]: moving to ovmm_task.py
-            spawn_recs = [
-                sim.receptacles[
-                    episode.name_to_receptacle[
-                        list(sim.instance_handle_to_ref_handle.keys())[0]
-                    ]
-                ]
-            ]
-            snap_pos = np.array(
-                [r.get_surface_center(sim) for r in spawn_recs]
+        if self._episode_init:
+            self._sim.robot.base_pos = np.array(episode.start_position)
+            start_quat = quaternion_from_coeff(
+                episode.start_rotation
             )
-
+            direction_vector = np.array([0, 0, -1])
+            heading_vector = quaternion_rotate_vector(start_quat, direction_vector)
+            self._sim.robot.base_rot = cartesian_to_polar(-heading_vector[2], heading_vector[0])[1]
+        elif self._pick_init or self._place_init:
+            if self._pick_init:
+                spawn_goals = episode.candidate_objects
+            else:
+                spawn_goals = episode.candidate_goal_receps
+                # Remove whatever the agent is currently holding.
+                abs_obj_idx = sim.scene_obj_ids[self.abs_targ_idx]
+                sim.grasp_mgr.desnap(force=True)
+                sim.grasp_mgr.snap_to_obj(abs_obj_idx, force=True)
+            np.random.seed(42)
```

Comment:
What is this?

Reply:
removed. had added this for freezing place initializations for like-to-like place policy comparisons